driver/src/main.rs

```diff
@@ -0,0 +1,195 @@
+
+    let mut docs: Vec<models::Deposits> = cursor.map(|doc| doc.unwrap())
+                                .map(|doc| serde_json::to_string(&doc)
+                                    .map(|json| serde_json::from_str(&json).unwrap())
+                                        .expect("Failed to parse json")).collect();
+
+    docs.sort_by(|a, b| b.slot.cmp(&a.slot)); 
+    Ok(docs)
+}
+
+fn apply_deposits(state: &mut models::State, deposits: &Vec<models::Deposits>) -> Result<models::State, io::Error> {
+    for i in deposits {
+        state.balances[ (i.accountId * models::TOKENS + i.tokenId) as usize] += i.amount;
+    }
+    Ok(state.clone())
+}
+
+fn main() {
+    
+    let client = Client::connect("localhost", 27017)
+        .expect("Failed to initialize standalone client.");
+
+	let (_eloop, transport) = web3::transports::Http::new("http://localhost:8545").unwrap();
+	let web3 = web3::Web3::new(transport);
+    
+
+	// Reading contract abi from json file.
+	let contents = fs::read_to_string("./build/contracts/SnappBase.json")
+	    .expect("Something went wrong reading the SnappBasejson");
+	let  snapp_base: serde_json::Value  = serde_json::from_str(&contents).expect("Json convertion was not correct");
+	let  snapp_base_abi: String =snapp_base.get("abi").unwrap().to_string(); 
+
+	let address: Address  = Address::from("0xC89Ce4735882C9F0f0FE26686c53074E09B0D550");//Todo: read address .env
+	let contract = Contract::from_json(web3.eth(), address ,snapp_base_abi.as_bytes()) 
+	    .unwrap();
+
+    loop{
+	    let mut state = get_current_balances(client.clone()).expect("Could not get the current state of the chain");
+	    println!("Current balances are: {:?}", state.balances);
+
+
+	    let accounts = web3.eth().accounts().wait().unwrap();
+	    //Get current balance
+	    let balance = web3.eth().balance(accounts[0], None).wait().unwrap();
+	    if balance.is_zero() {
+	    	panic!("Not sufficient balance for posting updates into the chain: {}", balance);
+	    }
+
+
+		 //get depositSlot
+	   	let result = contract.query("depositSlot", (), None, Options::default(), None);
+	    let current_deposit_ind: U256 = result.wait().unwrap();
+	    let current_deposit_slot: i32 = current_deposit_ind.low_u32() as i32; 
+	    // get latest non-applied deposit_index 
+	    let mut deposit_ind: i32 = current_deposit_ind.low_u32() as i32 + 1;
+	    println!("Current top deposit_slot is {:?}", deposit_ind);
+	    let mut found: bool = false;
+
+	    // Starting from the last depositSlot, we search backwards to the first non-applied deposit
+	    while !found {
+	    	deposit_ind = deposit_ind - 1;
+	    	let result = contract.query("hasDepositBeenApplied", U256::from(deposit_ind), None, Options::default(), None);
+	    	found = result.wait().unwrap();
+	    	if deposit_ind == 0 {
+	    		break;
+	    	}
+	    }
+	    if found {
+	    	deposit_ind = deposit_ind + 1;
+	    }
+
+	    // Now, we want to hop through all empty depositSlots
```

Comment:
I promise to fix this ASAP! You won't have to do this for much longer!

Reply:
cool. That is great thanks